### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by cilfonegabriel**
# Dear Code Reviewers.
### Project requirements
- On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. `rentals.json`)
- On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

### Thank´s! Gabriel.

### CODE REVIEW

File: date/books.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[{"json_class":"Book","title":"Harry Potter","author":"Rowling"},{"json_class":"Book","title":"The Lord of the Rings","author":"Tolkien"},{"json_class":"Book","title":"kjbhjh","author":"kbjkhb"}]
```
Comment (julie-ify) [inline date/books.json]: - [ ] It appears you are not making use of the files within the `date` directory. When I create books, people, and rentals, the files are saved in JSON files within the root path. Kindly remove this file and all the ones within the `date` folder to keep your code clean.

File: date/people.json [hunk 2]
```diff
@@ -0,0 +1 @@
+[{"json_class":"Student","id":422,"name":"Gabriel","age":20,"parent_permission":true,"classroom":null},{"json_class":"Teacher","id":167,"name":"Vicente","age":30,"specialization":"vet"}]
```
Comment (julie-ify) [inline date/people.json]: - [ ] It appears you are not making use of the files within the `date` directory. When I create books, people, and rentals, the files are saved in JSON files within the root path. Kindly remove this file and all the ones within the `date` folder to keep your code clean.

File: date/rentals.json [hunk 3]
```diff
@@ -0,0 +1 @@
+[{"json_class":"Rental","date":"21/03","person":{"json_class":"Teacher","id":167,"name":"Vicente","age":30,"specialization":"vet"},"book":{"json_class":"Book","title":"Harry Potter","author":"Rowling"}},{"json_class":"Rental","date":"21/03","person":{"json_class":"Student","id":422,"name":"Gabriel","age":20,"parent_permission":true,"classroom":null},"book":{"json_class":"Book","title":"The Lord of the Rings","author":"Tolkien"}}]
```
Comment (julie-ify) [inline date/rentals.json]: - [ ] It appears you are not making use of the files within the `date` directory. When I create books, people, and rentals, the files are saved in JSON files within the root path. Kindly remove this file and all the ones within the `date` folder to keep your code clean.

File: modules/app.rb [hunk 4]
```diff
@@ -120,4 +124,42 @@ def list_rentals
...
+
+    JSON.parse(File.read('people.json')).map do |person|
+      if person['json_class'] == 'Student'
+        Student.new(person['age'], person['name'], person['parent_permission'])
+      elsif person['json_class'] == 'Teacher'
+        Teacher.new(person['age'], person['specialization'], name: person['name'])
+      end
+    end
+  end
+
```
Comment (julie-ify) [inline modules/app.rb]: - [ ] There seems to be a bug in your app. I finished creating a person and exited the app. When I rerun the app, it changes the `id` of the person on the JSON file. Kindly fix that. We need to persist the data of the person across multiple iterations without losing or updating the data.

  jfke

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @cilfonegabriel,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @cilfonegabriel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**